tests/fib/test_fib.py

```diff
@@ -67,41 +76,55 @@ def get_fib_info(duthost, cfg_facts, mg_facts):
             }
     """
     timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
-    duthost.shell("redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(timestamp))
-    duthost.fetch(src="/tmp/fib.{}.txt".format(timestamp), dest="/tmp/fib")
-
-    po = cfg_facts.get('PORTCHANNEL', {})
-    ports = cfg_facts.get('PORT', {})
-
     fib_info = {}
-    with open("/tmp/fib/{}/tmp/fib.{}.txt".format(duthost.hostname, timestamp)) as fp:
-        fib = json.load(fp)
-        for k, v in fib.items():
-            skip = False
-
-            prefix = k.split(':', 1)[1]
-            ifnames = v['value']['ifname'].split(',')
-            nh = v['value']['nexthop']
-
-            oports = []
-            for ifname in ifnames:
-                if po.has_key(ifname):
-                    oports.append([str(mg_facts['minigraph_ptf_indices'][x]) for x in po[ifname]['members']])
-                else:
-                    if ports.has_key(ifname):
-                        oports.append([str(mg_facts['minigraph_ptf_indices'][ifname])])
+    for asic_index, asic_cfg_facts in  enumerate(cfg_facts):
+
+        asic = duthost.asic_instance(asic_index)
+
+        asic.shell("{} redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(asic._ns_arg, timestamp))
+        duthost.fetch(src="/tmp/fib.{}.txt".format(timestamp), dest="/tmp/fib")
+
+        po = asic_cfg_facts.get('PORTCHANNEL', {})
+        ports = asic_cfg_facts.get('PORT', {})
+
+        with open("/tmp/fib/{}/tmp/fib.{}.txt".format(duthost.hostname, timestamp)) as fp:
+            fib = json.load(fp)
+            for k, v in fib.items():
+                skip = False
+
+                prefix = k.split(':', 1)[1]
+                ifnames = v['value']['ifname'].split(',')
+                nh = v['value']['nexthop']
+
+                oports = []
+                for ifname in ifnames:
+                    if po.has_key(ifname):
+                        # ignore the prefix, if the prefix nexthop is not a frontend port
+                        if 'members' in po[ifname]:
+                            if 'role' in ports[po[ifname]['members'][0]] and ports[po[ifname]['members'][0]]['role'] == 'Int':
+                                skip = True
```

Comment:
Are we checking if prefixes from T0 ASICs are learned on T1 ASICS, and vice versa? We are removing all the prefixes that have next hop as the backend.

Reply:
@smaheshm for this test purpose there is no difference if the route is learnt from T0 or T2.  We need to build the fib table with all the external nexthops for the ptf tests to validate the packet forwarding

Reply:
We use -64-lag topology for multi asic platform testing. In this topology some ASICs T2 neighbors and some have T0 neighbors.  This covers the case of more than one ASIC having the same prefix.